### PR TITLE
Add license dump option

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
1. Modifies the License and Rule files to have their data as YAML frontmatter.

* .RULE files and .LICENSE files now also contain their data.
* git history is preserved
* loading/dumping methods are modified to support YAML frontmatter
* License/Rule now has different methods to point to their file.

2. Adds a new command line option: `--get-license-data` to:

* Dump license data in JSON, YAML and HTML formats.
* Also dumps the .LICENSE file with text and data as YAML frontmatter.
* Generates an index and a static website to view the data.

This is reusing code originally located at:
https://github.com/nexB/scancode-licensedb

Reference: https://github.com/nexB/scancode-toolkit/issues/2738
Reference: https://github.com/nexB/scancode-toolkit/issues/3049

This also contains PR: https://github.com/nexB/scancode-toolkit/pull/3063 btw.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: src/licensedcode/models.py [hunk 1]
```diff
@@ -463,7 +463,7 @@ def load(self, license_file, check_consistency=True):
 
                 self.text = ''
             else:
-                self.text = post.content
+                self.text = post.content.lstrip()
```
Comment (pombredanne) [inline src/licensedcode/models.py]: Please do not lstrip all spaces, only new lines:
```
                self.text = post.content.lstrip("\n")
```

**Review by pombredanne — APPROVED** [review]
LGTM... Please do one last check wrt. spaces:
```Python
from licensedcode.models import *
gpl=db["gpl-2.0"]
bottom = gpl.text.strip("\n") + "\n"
top = saneyaml.dump(gpl.to_dict())
fm=f"---\n{top}\n---\n{bottom}"
```
IMHO redumping all licenses as frontmatter this way should be to help that we are not damaging space-based formatting.  In particualr watch out for the GPL -2.0.

File: src/licensedcode/models.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply src/licensedcode/models.py]: This is fixed now.

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne post your approval I've pushed the following changes:

1. Fix whitespace in License texts
2. Merge the latest changes in develop including external licenses work and make tests pass.
3. Update changelog

Should I go ahead and merge?

**Comment by AyanSinhaMahapatra** [comment]
Some scripts  need update but I opened an issue: #3138 to track that separately to not block this anymore.

**Comment by AyanSinhaMahapatra** [comment]
Merging this!

**Comment by AyanSinhaMahapatra** [comment]
Note that the command line option was later renamed from `--get-license-data` to `--dump-license-data` later.

**Pull request merged**